packages/client/src/redux/document.ts

```diff
@@ -0,0 +1,47 @@
+import { createSelector, createSlice } from "@reduxjs/toolkit";
+import type { RootState } from "./store";
+
+interface IDocHash {
+  [docId: string]: { id: string; creator_id: string };
+}
+
+interface DocumentState {
+  doc?: { id: string; creator_id: string };
+  userDocs?: IDocHash;
+}
+
+const initialState: DocumentState = {};
+
+export const docSlice = createSlice({
```

Comment:
there is a lot happening in this commit already, maybe splitting up the changes would make sense? Are there any :atom: ideas that can be extracted?